### PR TITLE
Add test for pausing a sequence

### PR DESCRIPTION
**Pull request by martiniil**
* The test checks that the motion fails, since pause of sequence is not implemented yet
* Closes gap in line coverage of robot.py

### CODE REVIEW

File: pilz_robot_programming/src/pilz_robot_programming/robot.py [hunk 1]
```diff
@@ -382,7 +382,8 @@ def _move_execution_loop(self, cmd):
                         if isinstance(cmd, Sequence):
                             rospy.logerr("Pause not implemented for sequence yet")
                             raise RobotMoveFailed("Pause not implemented for sequence yet")
-                        continue
+                        # the following line is never executed due to optimization
+                        continue # pragma: no cover
```
Comment (ct2034) [inline pilz_robot_programming/src/pilz_robot_programming/robot.py]: why is this line necessary? 
IMO the behaviour would be the same without

Comment (ct2034) [reply pilz_robot_programming/src/pilz_robot_programming/robot.py]: I tested it. This line can be removed and doesn't change anything (Which may be the reason why it is removed by optimization ;oD)

File: pilz_robot_programming/test/integrationtests/tst_api_pause.py [hunk 2]
```diff
@@ -500,6 +500,33 @@ def test_pause_and_new_command(self):
...
+        move_thread.start()
+        self.assertTrue(self.robot_motion_observer.wait_motion_start(
+            move_tolerance=self._TOLERANCE_FOR_MOTION_DETECTION_RAD, sleep_interval=self._SLEEP_TIME_S))
+
+        # 2. trigger pause
+        self.robot.pause()
+        self.assertTrue(self.robot_motion_observer.wait_motion_stop(self._WAIT_CMD_STOP_TIME_OUT_S,
+                                                                    self._TOLERANCE_FOR_MOTION_DETECTION_RAD))
+        move_thread.join()
+        self.assertTrue(move_thread.exception_thrown)
```
Comment (ct2034) [inline pilz_robot_programming/test/integrationtests/tst_api_pause.py:528]: I noticed that this exception is only thrown for the first time you call `pause()`. Is this the expected behaviour?

**Review by ct2034 — CHANGES_REQUESTED** [review]

File: pilz_robot_programming/test/integrationtests/tst_api_pause.py (hunk 2 above)
Comment (martiniil) [reply pilz_robot_programming/test/integrationtests/tst_api_pause.py:528]: Yes. The sequence is aborted when you try to pause it. A second `pause()` has no effect.

File: pilz_robot_programming/src/pilz_robot_programming/robot.py (hunk 1 above)
Comment (martiniil) [reply pilz_robot_programming/src/pilz_robot_programming/robot.py]: The effect is that execution jumps directly to the beginning of the while loop. But you are right, it is not necessary.

**Review by ct2034 — APPROVED** [review]
